lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -0,0 +1,766 @@
+       It uses the pyOpenSSL python library to interact with OpenSSL."
+requirements:
+    - "python-pyOpenSSL >= 0.15"
+    - acme-tiny (if using the acme provider)
+options:
+    state:
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the certificate should exist or not, taking action if the state is different from what is stated.
+    path:
+        required: true
+        description:
+            - Remote absolute path where the generated certificate file should be created or is already located.
+    provider:
+        required: true
+        choices: [ 'selfsigned', 'assertonly', 'acme' ]
+        description:
+            - Name of the provider to use to generate/retrieve the OpenSSL certificate.
+              The 'assertonly' provider will not generate files and fail if the certificate file is missing.
+    force:
+        default: False
+        choices: [ True, False ]
+        description:
+            - Generate the certificate, even if it already exists.
+    csr_path:
+        description:
+            - Path to the Certificate Signing Request (CSR) used to generate this certificate. This is not required in 'assertonly' mode.
+
+    privatekey_path:
+        description:
+            - Path to the private key to use when signing the certificate.
+
+    privatekey_passphrase:
+        description:
+            - The passphrase for the privatekey.
+
+    selfsigned_digest:
+        default: "sha256"
+        description:
+            - Digest algorithm to be used when self-signing the certificate
+
+    selfsigned_notBefore:
+        description:
+            - The timestamp at which the certificate starts being valid. The timestamp is formatted as an ASN.1 TIME.
+              If this value is not specified, certificate will start being valid from now.
+
+    selfsigned_notAfter:
+        description:
+            - The timestamp at which the certificate stops being valid. The timestamp is formatted as an ASN.1 TIME.
+              If this value is not specified, certificate will stop being valid 10 years from now.
+
+    acme_accountkey:
+        description:
+            - Path to the accountkey for the 'acme' provider
+
+    acme_challenge_path:
+        description:
+            - Path to the ACME challenge directory that is served on http://<HOST>:80/.well-known/acme-challenge/
+
+    signature_algorithms:
+        description:
+            - list of algorithms that you would accept the certificate to be signed with
+              (e.g. ['sha256WithRSAEncryption', 'sha512WithRSAEncryption']).
+
+    issuer:
+        description:
+            - Key/value pairs that must be present in the issuer name field of the certificate
+
+    subject:
+        description:
+            - Key/value pairs that must be present in the subject name field of the certificate
+
+    has_expired:
+        choices: [ True, False ]
```

Comment:
Boolean parameters: add default value (`False` is default value for `has_expired`, `extendedKeyUsage_strict`, `subjectAltName_strict`).